app/src/main/java/org/cru/godtools/activity/BasePlatformActivity.java

```diff
@@ -242,16 +252,58 @@ private void setupNavigationDrawer() {
         }
     }
 
+    /**
+     * This method is used to update the Navigation Draws Login settings.
+     * It will only show log in information if the device is set to english
+     * and display log in links based on users status.
+     * Updated by:  Gyasi Story
+     */
     private void updateNavigationDrawerMenu() {
-        if (mLoginItem != null) {
-            mLoginItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
-        }
-        if (mSignupItem != null) {
-            mSignupItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
+        if (!containsLoginLanguage()) {  // For non English
+            if (mLoginItem != null) {
+                mLoginItem.setVisible(false);
+            }
+            if (mSignupItem != null) {
+                mSignupItem.setVisible(false);
+            }
+            if (mLogoutItem != null) {
+                mLogoutItem.setVisible(false);
+            }
+
+        } else { // For English
+            Boolean isNotLoggedIn = mTheKey.getDefaultSessionGuid() == null;
+            if (mLoginItem != null) {
+                mLoginItem.setVisible(isNotLoggedIn);
+            }
+            if (mSignupItem != null) {
+                mSignupItem.setVisible(isNotLoggedIn);
+            }
+            if (mLogoutItem != null) {
+                mLogoutItem.setVisible(!isNotLoggedIn);
+            }
         }
-        if (mLogoutItem != null) {
-            mLogoutItem.setVisible(mTheKey.getDefaultSessionGuid() != null);
+    }
+
+    /**
+     * This method will determine if the application contains
+     * a supported language for Login Notifications. As we
+     * get more support for more languages you can add them to {@code languages}.
+     *
+     * @return Boolean value if login Language
+     * <p>
+     * Updated by: Gyasi Story
+     */
+    private boolean containsLoginLanguage() {
+        LocaleListCompat localeListCompat = ConfigurationCompat
+                .getLocales(Resources.getSystem().getConfiguration());
+        String matchLanguage = localeListCompat.getFirstMatch(mLoginLanguages).getLanguage();
+        for (String lang : mLoginLanguages) {
+            if (matchLanguage.equals(lang)) {
+                return true;
+            }
         }
```

Comment:
@frett I changed the loop so it will only iterate through one collection instead of two.  Let me know if you are OK with this before I merge to development (didn't realize you already approved).